firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -169,6 +191,15 @@ private static boolean persistedFidMissingOrInErrorState(PersistedFidEntry persi
     };
   }
 
+  @NonNull
+  private <F, T> Continuation<F, T> call(@NonNull Supplier<T> supplier) {
+    return t -> {
+      // Waiting for Task that registers FID on the FIS Servers
+      executor.awaitTermination(1000, TimeUnit.MILLISECONDS);
```

Comment:
this is just 1, TimeUnit.SECONDS

Reply:
I don't think `awaitTermination` is right here unless you're shutting down the executor service somewhere else. From the JavaDoc: "Blocks until all tasks have completed execution after a shutdown request, or the timeout occurs, or the current thread is interrupted, whichever happens first."

Reply:
you shouldn't need to do anything like that - then continueWith part ensures that this only gets called when the previous task is completed

Reply:
But we have a background task that registers the FID on FIS Servers, which runs independent of the prev task. That's why I wanted to wait for sometime.

Reply:
Updated the code using CountDownLatch to await for the background task. 

@ciarand Wdyt?